src/api/v1/db/oracledb/doughs-dao.js

```diff
@@ -245,4 +245,22 @@ const postDough = async (body) => {
   }
 };
 
-export { getDoughs, postDough };
+/**
+ * Get the dough from the database with id `id`
+ *
+ * @param {number} id
+ * @returns {Promise<object>} the dough with id `id`
+ */
+const getDoughById = async (id) => {
+  const query = doughsGetQuery('ID = :id');
+  const bindParams = { id };
+  const connection = await getConnection();
+  try {
+    const { rows } = await connection.execute(query, bindParams);
+    return serializeDough(rows, `doughs/${id}`);
```

Comment:
Here is an example I got for `doughs/{id}`:

```json
{
    "links": {
        "self": "https://localhost/v1/doughs/1"
    },
    "data": [
        {
            "type": "dough",
            "id": "1",
            "links": {
                "self": "https://localhost/v1/doughs/1"
            },
            "attributes": {
                "name": "Sample dough",
                "gramsFlour": "500",
                "flourType": "Bread flour",
                "gramsWater": "400",
                "waterTemp": "90",
                "gramsYeast": "20",
                "gramsSalt": "10",
                "gramsSugar": "0",
                "gramsOliveOil": "50",
                "bulkFermentTime": "120",
                "proofTime": "30",
                "specialInstructions": "Can also proof overnight in the refrigerator"
            }
        }
    ]
}
```

A couple issues here:

1. Integer fields are still using strings.
2. `data` is an array instead of a object.

You probably wanna check your `/doughs` again as well.

Reply:
Yep, I'm seeing that in `/doughs` as well. I'm not sure why that is, I'll investigate.

Reply:
This fix fits right into the transform function. I'm surprised it's not something that the serializer can do automatically.

Reply:
This is how I transformed strings to floating numbers in Students API in case you need it: https://github.com/osu-mist/students-api/blob/defffa5442f163cedcb1f63b0c87ff17923256e3/src/api/v1/serializers/students-serializer.js#L79-L85

Reply:
Thanks, I like that idea of looping over a list of parameter names instead of writing each one out.